app/lib/text_to_chunk.rb

```diff
@@ -0,0 +1,29 @@
+class TextToChunk
+  include ActionView::Helpers::TagHelper
+  REGEXP_LIST_ITEM = /^(\s+)?(?<list_style_type>\d+\.|-)/
+
+  def chunk
+    lines.chunk do |line|
+      if line.blank?
+        :br
+      elsif REGEXP_LIST_ITEM.match?(line)
+        :li
+      else
+        :p
+      end
+    end.to_a
+  end
+
+  def lines
+    text.gsub(/\r\n?/, "\n") # replace CRLF by LF
+      .split(/\r|\n/) # split by CR/LF
```

Comment:
vu que t'as replace les \r, on doit pouvoir split \n uniquement. En fait on pourrait peut-être simplifier avec : 

```ruby
text.gsub(/[\r\n]+/, "\n").split("\n")

# sinon moins casse gueule
text.lines.filter_map { _1.strip.presence }
```

afin de supprimer les lignes vides et éviter les `<p></p>` vides ?

Reply:
> afin de supprimer les lignes vides et éviter les <p></p> vides ?
répondu ailleurs

> vu que t'as replace les \r, on doit pouvoir split \n uniquement. En fait on pourrait peut-être simplifier avec :
casse pied les CR/LF/CRLF. avec ton changement proposé

```
Finished in 0.04243 seconds (files took 1.33 seconds to load)
8 examples, 2 failures

Failed examples:

rspec ./spec/lib/text_to_chunk_spec.rb:18 # TextToChunk linerize (inputs might comes CR/LF/CRLF) many CR/LF/CRLF, only new line with content are split, otherwise keep empty items is expected to eq ["hello", "", "", "", "world"]
rspec ./spec/lib/text_to_chunk_spec.rb:45 # TextToChunk chunk with many CRLF? is expected to eq [[:p, ["hello"]], [:br, ["", "", ""]], [:p, ["world"]]]
```

mais trouvé qqch de mieux, \R [ttes sortes de nouvelles lignes]. merci pour ton feedback, j'en ai profité pr ajouter un test en multiligne